### PR TITLE
[13] Search user

### PR DESCRIPTION
**Pull request by moraesagustina**
image

image

image

### CODE REVIEW

File: app/assets/stylesheets/index-user.scss [hunk 1]
```diff
@@ -27,6 +27,32 @@
...
+    margin-top: 10%;
+
+    .text {
+      display: flex;
+      flex-direction: column;
+      align-items: center;
+      width: 150px;
+
+      p {
+        color: rgb(87, 87, 87);
```
Comment (lucianocaus) [inline app/assets/stylesheets/index-user.scss]: Let's use hexa. Nothing wrong with rgb, just to be consistent across the app.

File: app/models/user.rb [hunk 2]
```diff
@@ -44,6 +44,17 @@ def following?(other_user)
...
+  def self.search(query)
+    if query
+      User.where('LOWER(nickname) LIKE :query OR LOWER(first_name) LIKE :query OR LOWER(last_name) LIKE :query',
+                 query: "%#{query.downcase}%")
+    else
+      User.all
+    end
+  end
+
+  private_class_method :method
```
Comment (lucianocaus) [inline app/models/user.rb]: I think there's no need for this line

File: app/models/user.rb [hunk 3]
```diff
@@ -44,6 +44,17 @@ def following?(other_user)
...
   end
 
+  def self.search(query)
+    if query
+      User.where('LOWER(nickname) LIKE :query OR LOWER(first_name) LIKE :query OR LOWER(last_name) LIKE :query',
+                 query: "%#{query.downcase}%")
+    else
+      User.all
+    end
+  end
```
Comment (lucianocaus) [inline app/models/user.rb]: let's order them by username or some criteria

File: app/models/user.rb [hunk 4]
```diff
@@ -44,6 +44,17 @@ def following?(other_user)
     following.include?(other_user)
   end
 
+  def self.search(query)
```
Comment (lucianocaus) [inline app/models/user.rb]: I think you could move this directly to a scope

File: app/assets/stylesheets/index-user.scss (hunk 1 above)
Comment (moraesagustina) [reply app/assets/stylesheets/index-user.scss]: sorryy I forgot

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -22,7 +22,7 @@ def update
   end
 
   def index
-    @users = User.page(params[:page]).per(5)
+    @users = User.search(params[:query])&.ordered_by_first_name&.page(params[:page])&.per(5)
   end
```
Comment (lucianocaus) [inline app/controllers/users_controller.rb]: Include the ordered on the search itself

**Review by lucianocaus — APPROVED** [review]
Approved, fan del patito

**Pull request merged**